lint: introduce lint check with pylint

fernflower opened the pull request:
Added a naive pylint config with everything
disabled.
The following commits will enable a checking
option following a necessary code refactor.

vojtechsokol commented on Makefile:
```diff
@@ -125,4 +125,7 @@ test:
+pep8:
+	pylint leapp tests/scripts/*.py
+
 .PHONY: clean copr_build install install-deps install-test srpm test
```
```suggestion
.PHONY: clean copr_build install install-deps install-test srpm test pep8
```

fernflower replied:
Sure, I'll remove tox.ini and fix phony targets list.
As for using pytest-pylint - I spent around 1 hour trying to make it not ignore anything but tests and failed, so I returned to the pure pylint. Maybe you have the idea what I was doing wrong and why py.test --pylint was ignoring project directory skipping all files there?

vojtechsokol replied:
The trick is to pass path to pylint, otherwise it searches for tests. So `py.test --pylint --flake8 leapp` works for me, it invokes pylint and flake8 checks on all `*.py` files in the `leapp` folder. Also if pytest is used for invoking linters, all configuration can be stored in one file, for both tests, and linters - `setup.cfg`. So even `.flake8` config could be deleted then.

fernflower commented:
@vojtechsokol Removed tox.ini and flake8 configs, switched to pytest-pylint and refactored the Makefile as you suggested. Also couldn't help it and sorted alphabetically the dependencies in requirements-test =)

vojtechsokol commented on leapp/cli/__main__.py:
```diff
@@ -1,4 +1,3 @@
-import leapp.utils.i18n
```
Are you sure this import should be removed? Sure it's unused in this file, but looking at the [PR](https://github.com/oamg/leapp/pull/364/files#diff-23d9ccc8e4f98ba7963856040ff788b2) where this line was introduced, it was unused even back then, so maybe this is intentional.

vinzenz replied:
Indeed this is intentional to setup the builtins overrides for _ and P_

fernflower replied:
Good catch, thanks. Was on a killing spree, will fix

fernflower replied:
As for setup.cfg we should keep [tool:pytest] section if we intend to keep flake8 check, otherwise
`pytest --cache-clear --flake8 -m flake8 leapp tests/scripts/*.py` will produce
```=================================== FAILURES ===================================
_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/leapp/cli/report/__init__.py:11:1: F401 'leapp.utils.clicmd.command_arg' imported but unused

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/leapp/dialogs/__init__.py:1:1: F401 'leapp.dialogs.components.NumberComponent' imported but unused
/home/ivasilev/projects/leapp/leapp/dialogs/__init__.py:1:1: F401 'leapp.dialogs.components.BooleanComponent' imported but unused
/home/ivasilev/projects/leapp/leapp/dialogs/__init__.py:1:1: F401 'leapp.dialogs.components.ChoiceComponent' imported but unused
/home/ivasilev/projects/leapp/leapp/dialogs/__init__.py:1:1: F401 'leapp.dialogs.components.MultipleChoiceComponent' imported but unused
/home/ivasilev/projects/leapp/leapp/dialogs/__init__.py:4:1: F401 'leapp.dialogs.renderer.CommandlineRenderer' imported but unused

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/leapp/libraries/stdlib/__init__.py:9:1: F401 'subprocess' imported but unused
/home/ivasilev/projects/leapp/leapp/libraries/stdlib/__init__.py:12:1: F401 'six' imported but unused

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/leapp/messaging/__init__.py:8:1: F401 'six.moves.configparser' imported but unused

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/leapp/reporting/__init__.py:1:1: F401 'json' imported but unused

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_messaging.py:29:29: F811 redefinition of unused 'repository_dir' from line 8
/home/ivasilev/projects/leapp/tests/scripts/test_messaging.py:40:18: F811 redefinition of unused 'repository_dir' from line 8
/home/ivasilev/projects/leapp/tests/scripts/test_messaging.py:66:23: F811 redefinition of unused 'repository_dir' from line 8
/home/ivasilev/projects/leapp/tests/scripts/test_messaging.py:75:26: F811 redefinition of unused 'repository_dir' from line 8

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_repository_actor_definition.py:23:27: F811 redefinition of unused 'repository_dir' from line 6

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_repository.py:33:16: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_repository.py:81:15: F811 redefinition of unused 'repository_dir' from line 7

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:16:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:45:18: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:52:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:62:47: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:75:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:100:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:125:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:152:23: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:173:23: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:199:20: F811 redefinition of unused 'repository_dir' from line 7
/home/ivasilev/projects/leapp/tests/scripts/test_snactor.py:215:25: F811 redefinition of unused 'repository_dir' from line 7

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_stdlib_call.py:86:1: F811 redefinition of unused 'test_callability_check' from line 80

_________________________________ FLAKE8-check _________________________________
/home/ivasilev/projects/leapp/tests/scripts/test_utils_project.py:21:30: F811 redefinition of unused 'repository_dir' from line 6
/home/ivasilev/projects/leapp/tests/scripts/test_utils_project.py:58:34: F811 redefinition of unused 'repository_dir' from line 6
/home/ivasilev/projects/leapp/tests/scripts/test_utils_project.py:65:34: F811 redefinition of unused 'repository_dir' from line 6
```

vojtechsokol commented on Makefile:
```diff
@@ -121,8 +121,10 @@ install-test:
 
-.PHONY: clean copr_build install install-deps install-test srpm test
+lint:
+	py.test --cache-clear --flake8 --pylint -m pylint leapp tests/scripts/*.py
```
```suggestion
	py.test --cache-clear --flake8 --pylint -m flake8,pylint leapp tests/scripts/*.py
```

fernflower replied:
Having 2 modules `-m pylint,flake8` causes an unpleasant side-effect of running unit tests together with linter. I'd honestly prefer to have only linter checks under lint target.

vojtechsokol commented on .pylintrc:
```diff
@@ -0,0 +1,67 @@
+
+[FORMAT]
+# Maximum number of characters on a single line.
+max-line-length=120
```
Maybe change this to 119 to be consistent with [leapp-repository](https://github.com/oamg/leapp-repository/pull/211#pullrequestreview-242207314)?

fernflower replied:
I left 120 on purpose as changing it to 119 instantly shows that existing code base has violated this rule many times. imho 120->119 is a change for the next patch.
`=================================== FAILURES ===================================
______________________ [pylint] leapp/actors/__init__.py _______________________
C:308, 0: Line too long (120/119) (line-too-long)
C:313, 0: Line too long (120/119) (line-too-long)
C:324, 0: Line too long (120/119) (line-too-long)
C:355, 0: Line too long (120/119) (line-too-long)
C:378, 0: Line too long (120/119) (line-too-long)
_______________________ [pylint] leapp/dialogs/dialog.py _______________________
C: 55, 0: Line too long (120/119) (line-too-long)
___________________ [pylint] leapp/models/fields/__init__.py ___________________
C:447, 0: Line too long (120/119) (line-too-long)
____________________ [pylint] leapp/repository/__init__.py _____________________
C:  9, 0: Line too long (120/119) (line-too-long)
_________________ [pylint] leapp/snactor/commands/discover.py __________________
C:102, 0: Line too long (120/119) (line-too-long)
C:123, 0: Line too long (120/119) (line-too-long)
_______________________ [pylint] leapp/tags/__init__.py ________________________
C: 24, 0: Line too long (120/119) (line-too-long)
___________________________ test_audit_command_in_db ___________`
